packages/eslint-plugin-vtex/lib/rules/prefer-use-effect-named-callback.js

```diff
@@ -0,0 +1,77 @@
+const { getDocUrl } = require('../includes/getDocUrl.js')
+
+const USE_EFFECT_NAME = 'useEffect'
+
+const meta = {
+  docs: {
+    description: 'Prefer useEffect with named function or constant callbacks.',
+    category: 'Best Practices',
+    recommended: true,
+    uri: getDocUrl('prefer-use-effect-named-callback'),
+  },
+}
+
+function checkCallExpressionIsUseEffect(node) {
+  if (!node.callee) {
+    return false
+  }
+
+  // verify this scenario `useEffect(...)`
+  if (
+    node.callee.type === 'Identifier' &&
+    node.callee.name === USE_EFFECT_NAME
+  ) {
+    return true
+  }
+
+  // verify this scenario `React.useEffect(...)`
+  if (
+    node.callee.type === 'MemberExpression' &&
+    node.callee.property &&
+    node.callee.property.type === 'Identifier' &&
+    node.callee.property.name === USE_EFFECT_NAME
+  ) {
+    return true
+  }
+
+  return false
+}
+
+function create(context) {
+  function checkUseEffectCallExpression(cexpNode) {
+    if (!checkCallExpressionIsUseEffect(cexpNode)) {
+      return
+    }
+
+    cexpNode.arguments.forEach(function checkArgumentHasName(argument) {
```

Comment:
Shouldn't you just check the first argument? Seems unnecessary to check all the arguments here.